op-supervisor/supervisor/backend/db/db.go

```diff
@@ -135,6 +135,36 @@ func (db *DB) init() error {
 	return nil
 }
 
+func (db *DB) trimInvalidTrailingEntries() error {
+	i := db.lastEntryIdx()
+	for ; i >= 0; i-- {
+		entry, err := db.store.Read(i)
```

Comment:
Could go either way on it, but I'm saying it's not valid because there's no reason to have a search checkpoint at the end of the file if there's no logs after it. That fits in with the way logs are added where it includes the search checkpoints in the batch of things added for a sepecific log - if we roll back that log we should roll back the search checkpoints as well.  It also means that if you have `<init msg>, <checkpoint>, <exec link>` at the end of the file, we need to truncate to before that init msg so it's simplest if we just don't accept trailing checkpoints - otherwise we need to conditionally look at what came before them to know if we should truncate them or not.

Reply:
SGTM